src/collision.cpp

```diff
@@ -241,13 +241,13 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 	/*
 		Calculate new velocity
 	*/
-	if (dtime > 0.5f) {
+	if (dtime > 2.5f) {
 		if (!time_notification_done) {
 			time_notification_done = true;
-			infostream << "collisionMoveSimple: maximum step interval exceeded,"
+			warningstream << "collisionMoveSimple: maximum step interval exceeded,"
 					" lost movement details!"<<std::endl;
 		}
-		dtime = 0.5f;
+		dtime = 2.5f;
```

Comment:
If the step dtime and the object's velocity is too high, the object can fly through obstacles, no?
I.e. suppose a dropped item is moving with vel.y=-1 node/s and dtime=2.5. In one step it can skip 2.5 nodes.
(Haven't tested.)

Reply:
Ok, tested.
I've used this to simulate 3s server lag:
`//lua local function f() local t0 = minetest.get_us_time() while minetest.get_us_time() - t0 < 3e6 do end  core.after(0, f) end f()`
I haven't managed to make a dropped item fly through a node.

Reply:
The logic is predictive, so it won't miss a collision.

The key is to remove non-contiguousness as much as possible, because that is a main reason why client and server get out of sync, which then leads to resets, which show as jumps/jitter.

I still think it's best to simply remove these limits completely, but I'll settle for these larger limits.